app/src/main/java/io/github/sumukhshiv/memberquiz/GameActivity.java

```diff
@@ -0,0 +1,347 @@
+                    editor.putInt("score", userScore); // Storing the new user score
+                    editor.commit(); //saving the changes
+                }
+                timer.cancel();
+                finish();
+                startActivity(getIntent());
+            }
+        });
+
+        buttonD.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                String buttonText = buttonD.getText().toString();
+                if (!contactName.equalsIgnoreCase(buttonText)) {
+                    Toast.makeText(getApplicationContext(), "Incorrect!", Toast.LENGTH_SHORT).show();
+                } else {
+                    int userScore = pref.getInt("score", 0); // getting last score from the SharedPrefs, if there is no previous score it sets it to 0
+                    userScore++;
+                    SharedPreferences.Editor editor = pref.edit();
+                    editor.putInt("score", userScore); // Storing the new user score
+                    editor.commit(); //saving the changes
+                }
+                timer.cancel();
+                finish();
+                startActivity(getIntent());
+            }
+        });
+
+        buttonEndGame.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                //timer.cancel();
+                //textViewTimer.setText("seconds remaining: VOID");
+                AlertDialog alertDialog = new AlertDialog.Builder(GameActivity.this).create();
+                alertDialog.setTitle("END GAME");
+                alertDialog.setMessage("Are you sure you want to quit?");
+                alertDialog.setButton(AlertDialog.BUTTON_POSITIVE, "YES",
+                        new DialogInterface.OnClickListener() {
+                            public void onClick(DialogInterface dialog, int which) {
+                                SharedPreferences.Editor editor = pref.edit();
+                                editor.putInt("score", 0); // Storing the new user score
+                                editor.commit(); //saving the changes
+                                Intent intent = new Intent(getApplicationContext(), MainActivity.class);
+                                startActivity(intent);
+                                dialog.dismiss();
+                            }
+                        });
+                alertDialog.setButton(AlertDialog.BUTTON_NEGATIVE, "NO",
+                        new DialogInterface.OnClickListener() {
+                            public void onClick(DialogInterface dialog, int which) {
+                                dialog.dismiss();
+                            }
+                        });
+                alertDialog.show();
+
+
+            }
+
+
+        });
+
+    }
+
+    @Override
+    public void onStop() {
+        timer.cancel();
+        super.onStop();
+    }
```

Comment:
Use onPause and onResume. onStop is not called every time you leave the Activity, which will lead to situations in which you leave the Activity and it reopens itself due to the intent at the end of the timer